parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/ParallelConsumerOptions.java

```diff
@@ -86,19 +86,62 @@ public enum ProcessingOrder {
      */
     public enum CommitMode {
 
+        // tag::transactionalJavadoc[]
         /**
-         * Periodically commits through the Producer using transactions. Slowest of the options, but no duplicates in
-         * Kafka guaranteed (message replay may cause duplicates in external systems which is unavoidable with Kafka).
+         * Periodically commits through the Producer using transactions.
          * <p>
-         * This is separate from using an IDEMPOTENT Producer, which can be used, along with {@link
-         * CommitMode#PERIODIC_CONSUMER_SYNC} or {@link CommitMode#PERIODIC_CONSUMER_ASYNCHRONOUS}.
+         * The benefits of using this mode are:
+         * <p>
+         * a) All records produced from a given source offset will be visible, or none will be
+         * ({@link org.apache.kafka.common.IsolationLevel#READ_COMMITTED}).
+         * <p>
+         * b) If any records making up a transaction have a terminal issue being produced, or the system crashes before
+         * finishing sending all the records and committing, none will ever be visible and the system will retry th e
+         * whole set in a new transaction.
+         * <p>
+         * c) a source offset, and it's produced records will be committed as a set. Normally: either the record
+         * producing could fail, or the committing of the source offset could fail, as they are individual operations.
+         * When uUsing Transactions, they are committed together - so if either operations fail, the transaction will
+         * never get committed, and upon recovery the system will retry the set again (and no duplicates will be visible
+         * in the topic).
+         * <p>
+         * Slowest of the options, but no duplicates in Kafka caused by producing a record from consuming a record, that
+         * has already had its offset committed (message replay may cause duplicates in external systems which is
+         * unavoidable).
+         * <p>
+         * Note that the system uses very large transactions (for transaction standards), by only committing by default
```

Comment:
Ah - this will cause massive stalls in reading the topic - min latency consumption of 5 seconds. System needs to reduce the frequency as does ks. Say default same asks? And encourage configuring.

Reply:
Highlight this as a disadvantage of using transactions for consumers at the head of the topic